### PR TITLE
Handle profile command

### PR DESCRIPTION
**Pull request by ldhnam**
Issue #7

### CODE REVIEW

File: handler/command.go [hunk 1]
```diff
@@ -42,6 +43,10 @@ func (h *CommandHandler) commands(c echo.Context) error {
 
 	user, err := h.userSvc.Find(s.UserID)
 	if err != nil {
+		if err == gorm.ErrRecordNotFound {
```
Comment (kurojs) [inline handler/command.go:46]: This should be, `errors.Is(err, gorm.ErrRecordNotFound)` cause the error has been wrapped

File: service/event/slack.go [hunk 2]
```diff
@@ -44,9 +48,19 @@ func (s *slackSvc) Verify(header http.Header, body []byte) (interface{}, error)
 	return slackevents.ParseEvent(json.RawMessage(body), slackevents.OptionNoVerifyToken())
 }
 
-func (s *slackSvc) Profile() error {
-	s.logger.Info("handling profile")
-	return nil
+func (s *slackSvc) Profile(channelID, userID string) error {
+	user, err := s.userSvc.Find(userID)
+	if err != nil {
+		if err == gorm.ErrRecordNotFound {
```
Comment (kurojs) [inline service/event/slack.go:54]: Check my previous comment ^

**Review by kurojs — APPROVED** [review]

File: service/event/slack.go [hunk 3]
```diff
@@ -18,20 +20,21 @@ type slackSvc struct {
 	logger         *zap.Logger
 	db             *gorm.DB
 	client         *slack.Client
+	userSvc        user.Service
 }
 
 // NewSlackService --
-func NewSlackService(logger *zap.Logger, db *gorm.DB, client *slack.Client) Service {
+func NewSlackService(logger *zap.Logger, db *gorm.DB, client *slack.Client, userSvc user.Service) Service {
```
Comment (lmquang) [inline service/event/slack.go:27]: e đừng import userSvc, vì nếu các svc import nhau sẽ bị lỗi import cycle, e xài db để query user

File: handler/command.go [hunk 4]
```diff
@@ -42,6 +44,10 @@ func (h *CommandHandler) commands(c echo.Context) error {
 
 	user, err := h.userSvc.Find(s.UserID)
 	if err != nil {
+		if errors.Is(err, gorm.ErrRecordNotFound) {
+			h.logger.Error("User does not exist", zap.Error(err), zap.String("user_id", s.UserID))
+			return c.NoContent(http.StatusNotFound)
+		}
 		h.logger.Error("cannot find user", zap.Error(err), zap.String("user_id", s.UserID))
 		return c.NoContent(http.StatusNotFound)
```
Comment (lmquang) [inline handler/command.go:52]: return Status internal server error

**Review by lmquang — CHANGES_REQUESTED** [review]

**Review by lmquang — APPROVED** [review]

**Pull request merged**